prov/sockets/src/sock_ep_msg.c

```diff
@@ -379,440 +378,196 @@ static int sock_ep_cm_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen)
-				cm_entry.fid = &sock_ep->ep.fid;
-				sock_ep_enable(&sock_ep->ep);
-
-				if (sock_eq_report_event(sock_ep->attr->eq,
-					FI_CONNECTED, &cm_entry,
-					sizeof(cm_entry), 0))
-					SOCK_LOG_ERROR("Error in writing to EQ\n");
-				break;
-
-			default:
-				break;
-			}
-			dlist_remove(entry);
-			free(msg_entry);
-			break;
-		}
-		entry = entry->next;
+		done += ret;
 	}
-	fastlock_release(&sock_ep->attr->cm.lock);
-}
-
-static void sock_pep_cm_handle_ack(struct sock_cm_entry *cm,
-				    struct sock_conn_hdr *hdr)
-{
-	struct sock_conn_hdr *msg_hdr;
-	struct dlist_entry *entry;
-	struct sock_cm_msg_list_entry *msg_entry;
-
-	fastlock_acquire(&cm->lock);
-	for (entry = cm->msg_list.next; entry != &cm->msg_list;) {
-		msg_entry = container_of(entry, struct sock_cm_msg_list_entry,
-					  entry);
-		msg_hdr = (struct sock_conn_hdr *) msg_entry->msg;
-
-		if (msg_hdr->msg_id == hdr->msg_id) {
-			switch (msg_hdr->type) {
-			case SOCK_CONN_REJECT:
-				SOCK_LOG_DBG("Got ack for SOCK_CONN_REJECT\n");
-				break;
-
-			default:
-				SOCK_LOG_DBG("Invalid event: %d\n", msg_hdr->type);
-				break;
-			}
-			dlist_remove(entry);
-			free(msg_entry);
-			break;
-		}
-		entry = entry->next;
-	}
-	fastlock_release(&cm->lock);
+	return 0;
 }
 
-static void sock_release_shutdowns(struct sock_cm_entry *cm)
+static void sock_ep_cm_prepare_hdr(struct sock_conn_hdr *hdr)
 {
-	struct sock_conn_hdr *msg_hdr;
-	struct dlist_entry *entry, *next_entry;
-	struct sock_cm_msg_list_entry *msg_entry;
-
-	fastlock_acquire(&cm->lock);
-	for (entry = cm->msg_list.next; entry != &cm->msg_list;) {
-		msg_entry = container_of(entry, struct sock_cm_msg_list_entry,
-					  entry);
-		msg_hdr = (struct sock_conn_hdr *) msg_entry->msg;
-		next_entry = entry->next;
-
-		if (msg_hdr->type == SOCK_CONN_SHUTDOWN) {
-			SOCK_LOG_DBG("Discarding previous SOCK_CONN_SHUTDOWN in disconnecting state\n");
-			dlist_remove(entry);
-			free(msg_entry);
-		}
-		entry = next_entry;
-	}
-	fastlock_release(&cm->lock);
+	hdr->port = htons(hdr->port);
+	hdr->cm_data_sz = htonl(hdr->cm_data_sz);
 }
 
-static int sock_is_connecting(struct sock_cm_entry *cm,
-				struct sock_conn_hdr *hdr)
+static void sock_ep_cm_process_hdr(struct sock_conn_hdr *hdr)
 {
-	struct sock_conn_hdr *msg_hdr;
-	struct dlist_entry *entry;
-	struct sock_cm_msg_list_entry *msg_entry;
-
-	fastlock_acquire(&cm->lock);
-	for (entry = cm->msg_list.next; entry != &cm->msg_list; entry = entry->next) {
-		msg_entry = container_of(entry, struct sock_cm_msg_list_entry,
-					  entry);
-		msg_hdr = (struct sock_conn_hdr *) msg_entry->msg;
-
-		if (((msg_hdr->type == SOCK_CONN_REQ) || (msg_hdr->type == SOCK_CONN_ACK))
-		     && (hdr->msg_id < msg_hdr->msg_id)) {
-			SOCK_LOG_DBG("Discarding SOCK_CONN_SHUTDOWN in connecting state\n");
-			fastlock_release(&cm->lock);
-			return 1;
-		}
-	}
-	fastlock_release(&cm->lock);
-	return 0;
+	hdr->port = ntohs(hdr->port);
+	hdr->cm_data_sz = ntohl(hdr->cm_data_sz);
```

Comment:
I understand what you're trying to do, but having a structure that sometimes has some of the values in host order and sometimes some of the values in network order will be a pain to maintain.  I would drop these in favor of local variables in the places that need to read them.